src/controller/MahjongRound.ts

```diff
@@ -119,9 +78,8 @@ export class JapaneseRound {
 		for (const person of this.globalSeating) {
 			names.push(person.playerName);
 		}
-		const dealer = this.globalSeating[this.round - 1];
 
-		while (names[0] !== dealer.playerName) {
+		while (names[0] !== this.dealerName) {
 			names.push(names.shift());
 		}
 		for (const i in names) {
```

Comment:
I don't understand this code, do you mean to do something like:
```
for (let i = 0; i < names.length; i++) {
	this.localSeating[names[i]] = i;
}
```

Reply:
essentially the main purpose of this part is to transfer from [A0 B1 C2 D3] with C as the dealer to [C0 D1 A2 B3].